File: 2021/FileReader.py
class FileReader:

    def __init__(self):
        self.path = "resources/"

    def readLines(self, name):
        f = open(self.path + name + ".txt", "r")
        file = f.read()
        return file.split('\n')


    def readLinesAsInts(self, name):
        x = self.readLines(name)
        return [int(i) for i in x]

File: 2017/adventofcode/adventofcode/FileReader.cs
using System;
using System.IO;

namespace adventofcode
{
    public class FileReader {
        public static string ReadLine(string day)
        {
            try
            {
                using (var sr = new StreamReader("/usr/local/development/dotnet/adventofcode/adventofcode/resources/" + day + ".txt"))
                {
                    return sr.ReadToEnd();
                }
            }
            catch (IOException e)
            {
                Console.WriteLine("The file could not be read:");
                Console.WriteLine(e.Message);
            }
            return "";
        }

        public static List<string> ReadLines(string day)
        {
            List<string> result = new List<string>();
            try
            {
                using (var sr = new StreamReader("/usr/local/development/dotnet/adventofcode/adventofcode/resources/" + day + ".txt"))
                {
                    String? line;
                    while ((line = sr.ReadLine()) != null)
                    {
                        result.Add(line);
                    }
                }
            }
            catch (IOException e)
            {
                Console.WriteLine("The file could not be read:");
                Console.WriteLine(e.Message);
            }
            return result;
        }
    }
}

